### PR TITLE
Add check for params dhcp_subnets

### PR DESCRIPTION
**Pull request by andrei693**
Add new variable in params dhcp_subnets
Use validate_re to do a basic check if the IP addresses/masks are numbers
Set dhcp_subnets accordingly via template dhcp.yml.erb
Fixes GH-109

### CODE REVIEW

**Comment by domcleal** [comment]
Thanks for the pull request, a few comments:

1) The PR needs to be rebased against the current master (`git rebase master`), as it's changed since.
2) dhcp_subnets should probably be an array so multiple subnets can be specified. This might mean you need to remove the validate_re as I'm not sure there's an easy way to validate the contents of the array.  See cbda993f for an example of writing a template that generates an array in a slightly different YAML format, it's a bit easier.

**Comment by andrei693** [comment]
1) Did the rebase

2) The validate_re for $dhcp_subnets works with a comma separated list. See http://rubular.com/r/yb4leE78LK.

If this still does not suffice please let me know and I will look at adding it as an array.

**Comment by domcleal** [comment]
The PR doesn't appear to be rebased, GitHub reports it still conflicts.  All of the commits should also be squashed to a single commit.

I think I'd still prefer to pass an array of subnets into the parameter rather than a comma-separated string.

**Comment by andrei693** [comment]
My bad, git noob. Rebase is done now and I used an array. Unfortunately I don't have a 1.11 version around to test it.

File: manifests/init.pp [hunk 1]
```diff
@@ -171,6 +171,8 @@
 #
 # $dhcp_provider::              DHCP provider
 #
+# $dhcp_subnets::               DHCP subnets
```
Comment (domcleal) [inline manifests/init.pp]: missing type:array underneath here

Comment (domcleal) [inline manifests/init.pp]: If you could also expand the description, that would be very useful - e.g. "List of subnets to restrict DHCP management to"

**Comment by domcleal** [comment]
No problem, that's much tidier - a couple of inline comments left, but I tested it now and it works well.

**Comment by andrei693** [comment]
Done.

**Comment by domcleal** [comment]
:+1:

File: manifests/init.pp [hunk 2]
```diff
@@ -420,6 +424,9 @@
   validate_array($dhcp_option_domain)
   validate_integer($dhcp_omapi_port)
   validate_string($dhcp_provider, $dhcp_server)
+  if $dhcp_subnets {
```
Comment (ekohl) [inline manifests/init.pp]: This if statement isn't needed. It can always be an array (and it's the default).

**Comment by ekohl** [comment]
Looks good, but could you rephrase the commit message? The change is to expose the dhcp subnets parameter, not adding a check for it. Other than that it's good to go.

**Comment by andrei693** [comment]
Done, again. Removed the if statement too.

**Comment by ekohl** [comment]
Thanks!
